bitrepository-client/src/main/java/org/bitrepository/commandline/PutFileCmd.java

```diff
@@ -151,27 +170,58 @@ protected void validateArguments() {
      */
     private OperationEvent putTheFile() {
         output.debug("Uploading the file to the FileExchange.");
-        URL url = getURLOrUploadFile();
+        final URL url = getURLOrUploadFile();
+
         String fileID = retrieveFileID();
+        FileExchange fileExchange = ProtocolComponentFactory.getInstance().getFileExchange(settings);
 
         output.debug("Initiating the PutFile conversation.");
         ChecksumDataForFileTYPE validationChecksum = getValidationChecksum();
-        ChecksumSpecTYPE requestChecksum = getRequestChecksumSpecOrNull();
-
-        boolean printChecksums = cmdHandler.hasOption(Constants.REQUEST_CHECKSUM_TYPE_ARG);
+        ChecksumSpecTYPE requestedChecksumSpec = getRequestChecksumSpecOrDefault();
+
+        if (cmdHandler.hasOption(URL_ARG) && !cmdHandler.hasOption(PILLAR_ARG)) {
```

Comment:
The last part of this condition (`!cmdHandler.hasOption(PILLAR_ARG)`) can be removed. `put-file` doesn't expect `-p` so an exception would be thrown before this point if it were to be defined.